hazelcast/src/main/java/com/hazelcast/jet/impl/execution/init/ExecutionPlan.java

```diff
@@ -175,7 +176,11 @@ public void initialize(NodeEngineImpl nodeEngine,
         this.ptionArrgmt = new PartitionArrangement(partitionAssignment, nodeEngine.getThisAddress());
         Set<Integer> higherPriorityVertices = VertexDef.getHigherPriorityVertices(vertices);
         for (Address destAddr : remoteMembers.get()) {
-            memberConnections.put(destAddr, getMemberConnection(nodeEngine, destAddr));
+            Connection conn = getMemberConnection(nodeEngine, destAddr);
+            if (conn == null) {
+                throw new TopologyChangedException("no connection to job participant: " + destAddr);
```

Comment:
is this exception propagated back to the client?

Reply:
This staff is called on [InitExecutionOperation](https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/java/com/hazelcast/jet/impl/operation/InitExecutionOperation.java#L92-L99). since it is a jet [AsyncOperation](https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/java/com/hazelcast/jet/impl/operation/AsyncOperation.java#L50-L61) , it sends these exceptions as a response back to its caller (the job coordinator in this case).